internal/services/cognitive/cognitive_account_project_resource.go

```diff
@@ -0,0 +1,315 @@
+	DisplayName        string                                     `tfschema:"display_name"`
+	Endpoints          map[string]string                          `tfschema:"endpoints"`
+	Identity           []identity.ModelSystemAssignedUserAssigned `tfschema:"identity"`
+	IsDefault          bool                                       `tfschema:"is_default"`
+	Location           string                                     `tfschema:"location"`
+	Name               string                                     `tfschema:"name"`
+	Tags               map[string]string                          `tfschema:"tags"`
+}
+
+type CognitiveAccountProjectResource struct{}
+
+var _ sdk.ResourceWithUpdate = CognitiveAccountProjectResource{}
+
+func (r CognitiveAccountProjectResource) ResourceType() string {
+	return "azurerm_cognitive_account_project"
+}
+
+func (r CognitiveAccountProjectResource) ModelObject() interface{} {
+	return &CognitiveAccountProjectModel{}
+}
+
+func (r CognitiveAccountProjectResource) IDValidationFunc() pluginsdk.SchemaValidateFunc {
+	return cognitiveservicesprojects.ValidateProjectID
+}
+
+func (r CognitiveAccountProjectResource) CustomizeDiff() sdk.ResourceFunc {
+	return sdk.ResourceFunc{
+		Func: func(ctx context.Context, metadata sdk.ResourceMetaData) error {
+			if metadata.ResourceDiff == nil {
+				return nil
+			}
+			// Azure API issue: description and display_name cannot be updated to empty values, see: https://github.com/Azure/azure-rest-api-specs/issues/38422
+			if oldDesc, newDesc := metadata.ResourceDiff.GetChange("description"); oldDesc.(string) != "" && newDesc.(string) == "" {
+				metadata.ResourceDiff.ForceNew("description")
+			}
+
+			if oldDisplay, newDisplay := metadata.ResourceDiff.GetChange("display_name"); oldDisplay.(string) != "" && newDisplay.(string) == "" {
+				metadata.ResourceDiff.ForceNew("display_name")
+			}
+
+			return nil
+		},
+	}
+}
+
+func (r CognitiveAccountProjectResource) Arguments() map[string]*schema.Schema {
+	return map[string]*schema.Schema{
+		"name": {
+			Type:     pluginsdk.TypeString,
+			Required: true,
+			ForceNew: true,
+			ValidateFunc: validation.StringMatch(
+				regexp.MustCompile("^[a-zA-Z0-9][a-zA-Z0-9_.-]{1,63}$"),
+				"name must be between 2 and 64 characters long, start with an alphanumeric character, and contain only alphanumeric characters, dashes(-), periods(.) or underscores(_).",
+			),
+		},
+
+		"cognitive_account_id": commonschema.ResourceIDReferenceRequiredForceNew(&cognitiveservicesprojects.AccountId{}),
+
+		"location": commonschema.Location(),
+
+		"description": {
+			Type:         pluginsdk.TypeString,
+			Optional:     true,
+			ValidateFunc: validation.StringIsNotEmpty,
+		},
+
+		"display_name": {
+			Type:         pluginsdk.TypeString,
+			Optional:     true,
+			ValidateFunc: validation.StringIsNotEmpty,
+		},
+
+		"identity": commonschema.SystemAssignedUserAssignedIdentityRequired(),
```

Comment:
`required` properties should be put ahead of `optional` ones.

Reply:
fixed